[skip circleci] Add modern issue templates (#2394)
haskell/haskell-language-server: .github/ISSUE_TEMPLATE/bug_report.md

---
name: Bug report
about: Create a report to help us improve
title: ''
labels: 'status: needs triage, type: bug'
assignees: ''

---

<!--
If you encounter a bug or you have a support question, please try to fill out some of the information below.

> Please take a look to the [troubleshooting guide](https://haskell-language-server.readthedocs.io/en/latest/troubleshooting.html) before filling this new one.

Reviewer: ```suggestion
Please take a look at the [troubleshooting guide](https://haskell-language-server.readthedocs.io/en/latest/troubleshooting.html) before filing a new issue.
```

---
name: Bug report
about: Create a report to help us improve
title: ''
labels: 'status: needs triage, type: bug'
assignees: ''

---

<!--

> If you encounter a bug or you have a support question, please try to fill out some of the information below.

Reviewer 2: For some reason it won't let me make a suggestion here, but here's what I suggest:
```
Before opening an issue, please take a look at the [troubleshooting guide](https://haskell-language-server.readthedocs.io/en/latest/troubleshooting.html). This explains some common issues and will also help you to find the information that the issue template asks for.

When filing an issue, please fill out as much of the information below as you can. This helps us to debug your issue, but is not required!
```

Author: done!

Generally speaking, the information below is meant to help debugging issues but is no prerequisite for opening an issue.
Please take a look at the [troubleshooting guide](https://haskell-language-server.readthedocs.io/en/latest/troubleshooting.html) before filing a new issue.
The information below is meant to help debugging issues but is no prerequisite for opening an issue.
-->

### Your environment

> Output of `haskell-language-server --probe-tools` or `haskell-language-server-wrapper --probe-tools`:

Reviewer 2: This is still useful, no?

Reviewer: `haskell-language-server --debug .` also prints the tools, so I think this is indeed superfluous.

<!-- Tell us what happens instead. -->

### Include debug information
Execute in the root of your project the command `haskell-language-server-wrapper --debug .` and paste the logs here:

> (if you are using the vscode extension check the executable location [here](https://github.com/haskell/vscode-haskell#downloaded-binaries))

Reviewer 2: I wonder if we should just write more about this on the troubleshooting page, and then here we could say "If you don't know where to find the `haskell-language-server` exectuable, consult the troubleshooting page"

Author: Yeah, although a direct reference to the location may be helpful too? I mean, we are linking the troubleshooting page in the beginning but in this point the focus is locate it and the table linked gives you the path

```
</details>

> Paste the logs from the lsp-client, you can check instructions about for VS Code [here](https://github.com/haskell/vscode-haskell#troubleshooting)

Reviewer 2: Again, I think we should refer people to the relevant section of the troubleshooting page.

Author: hmm https://github.com/haskell/vscode-haskell#troubleshooting would not be the relevant page?
